File: CoreAssessment-frontend/src/Model/Admin/orgChart.ts
import { Grid } from '@mui/material';
import { dept } from './../../constant/authen';
export interface OrgChartDept {
    dept_cd?:string;
    dept_name?:string;
    dept_persons?:OrgChartEmployeeApprove[]

  }


   export interface OrgChartEmployeeApprove {

    dept_person?:string;
    sections?:OrgChartSection[];

  }

  export interface OrgChartSection {
    sect_name?:string;
    sect_cd?:string;
    sect_person?:string;
    grps?:OrgChartGroup[];

  }
  

  export interface OrgChartGroup {
    grp_cd?:string;
    grp_name?:string;
    grp_person?:string;


  }

  export interface Depts {
    dept_cd?:string;
    dept_name?:string;
    sects:Sects[]
  }

  export interface Sects {
    sect_cd?:string;
    sect_name?:string;
    groups:Groups[]
  }


  export interface Groups {
    grp_cd?:string;
    grp_name?:string;
  }


  

File: CoreAssessment-backend/CoreAssessment-backend/Model/OrgChart/OrgChart.cs
using CoreAssessment_backend.Model.DashboardBarCharts;

namespace CoreAssessment_backend.Model.RuleTree
{
    public class OrgChart
    {
       
    }

    public class OrgChartDept
    {
        public string? dept_cd { get; set; }

        public string? dept_name { get; set; }


        public List<OrgChartEmployeeApprove>? dept_persons { get; set; }
    }

    public class OrgChartEmployeeApprove
    {
        public string? dept_person { get; set; }

        public List<OrgChartSection>? sections { get; set; }
    }


    public class OrgChartSection
    {
        public string? sect_name { get; set; }

        public string? sect_cd { get; set; }
        public string? sect_person { get; set; }

        public List<OrgChartGroup>? grps { get; set; }
    }

    public class OrgChartGroup
    {

        public string? grp_cd { get; set; }
        public string? grp_name { get; set; }

        public string? grp_person { get; set; }



    }
}
